src/Showcase/src/Smart.FA.Catalog.Showcase.Web/Pages/Training/TrainingList.cshtml.cs

```diff
@@ -0,0 +1,43 @@
+#nullable disable
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.EntityFrameworkCore;
+using Smart.FA.Catalog.Shared.Domain.Enumerations.Training;
+
+namespace Smart.FA.Catalog.Showcase.Web.Pages.Training;
+
+public class TrainingListModel : PageModel
+{
+    private readonly Infrastructure.Data.CatalogShowcaseContext _context;
+
+    public TrainingListModel(Infrastructure.Data.CatalogShowcaseContext context)
+    {
+        _context = context;
+    }
+
+    public List<TrainingViewModel> Trainings { get; set; } = new List<TrainingViewModel>();
+
+    public async Task OnGetAsync()
+    {
+        var trainingList = await _context.TrainingList
+            .Where(training => training.TrainingStatus == TrainingStatus.Validated.Id)
+            .OrderBy(t => t.TrainingId)
+            .ToListAsync();
+
+        var trainingsByIds = trainingList.ToLookup(t => t.TrainingId);
+
+        foreach (var groupedTraining in trainingsByIds)
+        {
+            Trainings.Add(new TrainingViewModel()
+            {
+                TrainingId = groupedTraining.Key,
+                TrainingTitle = groupedTraining.First().TrainingTitle,
```

Comment:
If `groupedTraining` is empty, calling `First` will cause a crash.
Use `FirstOrDefault` instead and use the `?` operator.
Same remark for other instances where you use `First`.
```suggestion
                TrainingTitle = groupedTraining.FirstOrDefault()?.TrainingTitle,
```

Reply:
We do not call .First() at any time.

Reply:
We do not call .First() at any time for a training, We do it for the training status. Normally, there should be no trainings encoded without status. I can change it for .FirstOrDefault()

Reply:
Nah at that point we now for a fact we have at least one record.
We grouped by the training id just before.